File: phpdev/vs17/x64/main.py
import datetime

def test(n, p):
    if not p:
        return
    nn = range(0, n+1)
    for i in nn:
        ii = i * i
        for j in nn:
            jj = j * j
            for k in range(n-i-j, n-i-j+1):
                if i+j+k==n and ii+jj==k*k:
                    p(i, j, k)

def test_print(a, b, c):
    print "The result %d^2 + %d^2 = %d^2 !" % (a, b, c)

def main():
    starttime = datetime.datetime.now()

    test(1000, test_print)
    print "First Used %f seconds" % ((datetime.datetime.now() - starttime).seconds, )

    for i in range(0, 10):
        test(1000, None)

    print "All Used %f seconds" % ((datetime.datetime.now() - starttime).seconds, )

if __name__ == '__main__':
    main()

File: phpdev/vs17/x64/main.php
<?php

function test($n, $p)
{
	if (empty($p)) return;
    for ($i = 0; $i <= $n; $i++) {
	    $ii = $i * $i;
        for ($j = 0; $j <= $n; $j++) {
	        $jj = $j * $j;
            for ($k = $n-$i-$j; $k <= $n-$i-$j; $k++) {
                if ($i + $j + $k == $n && $ii + $jj == $k * $k) {
                    $p($i, $j, $k);
                }
            }
        }
    }
}

function main()
{
    $start = microtime(true);

    $test_print = function ($a, $b, $c) {
        echo "The result {$a}^2 + {$b}^2 = {$c}^2 !\n";
    };

    test(1000, $test_print);
    $cost1 = microtime(true) - $start;
    echo"First Used {$cost1} seconds\n";

    for ($i = 0; $i <= 10; $i++) {
        test(1000, null);
    }
    $cost = microtime(true) - $start;
    echo"All Used {$cost} seconds\n";
}


main();